### DISCUSSION TITLE
How to use cli export data?

### DISCUSSION THREAD

**Discussion by ghost** [openobserve/openobserve#4631] [Q&A]
I have a single node service deployed locally. There is some trace data, and exporting it using `openobserve.exe export --stream_type traces --data ./export_data` gives me an error.
```
D:\OpenObserve>openobserve.exe export --stream_type traces --data ./export_data
[2024-09-24T09:55:06Z INFO  infra::cache::file_data::disk] Loading disk cache start
[2024-09-24T09:55:06Z INFO  infra::cache::file_data::disk] Loading disk cache done, total files: 0
[2024-09-24T09:55:07Z INFO  openobserve::service::search] service:search:enter; trace_id="" org_id="default" stream_type=Traces user_id=None
[2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster::http] service:search:cluster; org_id="default"
[2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster] service:search:cluster:run; org_id="default"
[2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] is_final_phase: true
[2024-09-24T09:55:07Z ERROR openobserve::service::search::cluster] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] service:search:cluster:run: no querier node online
[2024-09-24T09:55:07Z ERROR openobserve::service::search::cluster::http] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] http->search: err: Message("no querier node online")
[2024-09-24T09:55:07Z INFO  openobserve::service::search] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] in leader task finish
search error: Message("no querier node online")
command export execute succeeded
```

**Comment by juju4** [comment]
I don't manage to get export working either.
From my ansible role CI
https://github.com/juju4/ansible-openobserve/actions/runs/12736612278/job/35496709754#step:16:18
```
+ sudo -H -u _openobserve env RUST_BACKTRACE=full /var/_openobserve/openobserve/openobserve export --stream journald
thread 'main' panicked at src/config/src/config.rs:1989:57:
create cache dir success: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
stack backtrace:
   0:     0x55e7ffb8883a - <unknown>
   1:     0x55e7fb1e89c3 - <unknown>
   2:     0x55e7ffb83c63 - <unknown>
   3:     0x55e7ffb88692 - <unknown>
   4:     0x55e7ffb89686 - <unknown>
   5:     0x55e7ffb894d0 - <unknown>
   6:     0x55e7ffb89c47 - <unknown>
   7:     0x55e7ffb89aea - <unknown>
   8:     0x55e7ffb88ce9 - <unknown>
   9:     0x55e7ffb8978c - <unknown>
  10:     0x55e7fb1e5a90 - <unknown>
  11:     0x55e7fb1e5ed6 - <unknown>
  12:     0x55e7fb1178fb - <unknown>
  13:     0x55e7fb07a317 - <unknown>
  14:     0x55e7fb110bec - <unknown>
  15:     0x55e7fd3708d3 - <unknown>
  16:     0x55e7fb110739 - <unknown>
  17:     0x55e7fef93809 - <unknown>
  18:     0x55e7fef92155 - <unknown>
  19:     0x55e7fe99d3b3 - <unknown>
  20:     0x55e7fe9984fd - <unknown>
  21:     0x55e7ffb7a619 - <unknown>
  22:     0x55e7fefab915 - <unknown>
  23:     0x7f2ee582a1ca - <unknown>
  24:     0x7f2ee582a28b - __libc_start_main
  25:     0x55e7f9e0d22e - <unknown>
  26:                0x0 - <unknown>
```

https://github.com/openobserve/openobserve/tree/main/src/config src/config/src/config.rs
memory limit issue? permission denied on what?
a more meaningful message would be nice :)

Thanks

**Comment by andrewmaspero** [comment]
> I have a single node service deployed locally. There is some trace data, and exporting it using `openobserve.exe export --stream_type traces --data ./export_data` gives me an error.
> 
> ```
> D:\OpenObserve>openobserve.exe export --stream_type traces --data ./export_data
> [2024-09-24T09:55:06Z INFO  infra::cache::file_data::disk] Loading disk cache start
> [2024-09-24T09:55:06Z INFO  infra::cache::file_data::disk] Loading disk cache done, total files: 0
> [2024-09-24T09:55:07Z INFO  openobserve::service::search] service:search:enter; trace_id="" org_id="default" stream_type=Traces user_id=None
> [2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster::http] service:search:cluster; org_id="default"
> [2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster] service:search:cluster:run; org_id="default"
> [2024-09-24T09:55:07Z INFO  openobserve::service::search::cluster] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] is_final_phase: true
> [2024-09-24T09:55:07Z ERROR openobserve::service::search::cluster] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] service:search:cluster:run: no querier node online
> [2024-09-24T09:55:07Z ERROR openobserve::service::search::cluster::http] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] http->search: err: Message("no querier node online")
> [2024-09-24T09:55:07Z INFO  openobserve::service::search] [trace_id 2mVpgKUASuTZQo4XLZBn29hIWGq] in leader task finish
> search error: Message("no querier node online")
> command export execute succeeded
> ```

I'm getting the exact same issue

**Comment by hengfeiyang** [comment]
This is a bug of this feature, it can't work for now. will fix it later.

**Reply by juju4** [reply]
Is there a workaround how to do backup? disk snapshot enough or specific dir backup?

**Reply by hengfeiyang** [reply]
If you are using single node, you need to backup:
1. Environment what you set
2. Data files in `/data/openobserve/streams/`
3. Metadata in `/data/openobserve/db/`

If you are using other database MySQL/Postgres, please also backup the database.
